service/controller/resource/certconfig/desired.go

```diff
@@ -37,26 +38,47 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	// Cluster deletion should not be affected only because some releases are
 	// missing or broken when fetching them from cluster-service.
 	if key.IsDeleted(&cr) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "not computing desired state of CertConfig CRs due to delete event")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "not computing desired state", "reason", "the current state is used for deletion")
 		return nil, nil
 	}
 
+	// We need to determine if we want to generate certificates for a Tenant
+	// Cluster with a HA Master setup.
+	var haMasterEnabled bool
+	{
+		haMasterEnabled, err = r.haMaster.Enabled(ctx, key.ClusterID(&cr))
```

Comment:
@tuommaki FYI I tried to not add a handler that works with the controller context.

Reply:
The design here is actually very good when considering that soon we might need to support different versions of `infrastructure` CRDs, depending on how things play out with AWS, Azure and our customers upgrade plans.